security/security-design/shared-assets/kms-backup/files/kms_key_backup.py

```diff
@@ -0,0 +1,489 @@
+    target_kms_service_endpoint = target_kms_vault_data.management_endpoint
+    target_kms_crypto_endpoint = target_kms_vault_data.crypto_endpoint
+    kms_key=key_data['kms_key_metadata']
+    exported_key=key_data['exported_key']
+    target_kms_crypto_client = oci.key_management.KmsCryptoClient(
+        config=target_kms_oci_config, service_endpoint=target_kms_crypto_endpoint
+    )
+    """
+    ###################################################
+    #### Get wrapping key from target vault
+    ###################################################
+    wrapping_key_raw = target_kms_management_client.get_wrapping_key().data.public_key
+    # strip new line characters from the key
+    wrapping_key = wrapping_key_raw.replace("\n", "")
+    """
+    ###################################################
+    #### Wrap the key for import
+    ###################################################
+    #current_key_data = target_kms_management_client.get_key(kms_key.id).data
+    current_key_data=key_data['current_key_data']
+    key_shape = oci.key_management.models.KeyShape(
+        algorithm=kms_key.algorithm,
+        length=int(current_key_data.key_shape.length),
+    )
+    wrapped_import_key = oci.key_management.models.WrappedImportKey(
+        key_material=exported_key.data.encrypted_key, wrapping_algorithm=wrapping_algorithm
+    )
+    ###################################################
+    #### Import the key to target
+    ###################################################
+    try:
+        import_key_details = oci.key_management.models.ImportKeyDetails(
+                            compartment_id=config['target_compartment'],
+                            display_name=config['target_keyname'],
+                            key_shape=key_shape,
+                            protection_mode=kms_key.protection_mode,
+                            wrapped_import_key=wrapped_import_key,
+                            freeform_tags={
+                                'source_vault_origin': config['target_vault'],
+                                'source_key_origin': kms_key.id,
+                            },
+        )
+        response = target_kms_management_client.import_key(
+            import_key_details
+        )
+        print("Imported key: " + config['target_keyname'] + " with ID: " + kms_key.id )
+        print("Target OCID " + response.data.id)
+        return(response)
+    except Exception as e:
+        print("Failed to import key " + kms_key.id + ": " + str(e))
+        return(False)
+
+
+
+#########################################################################
+# main
+# main function called from commandline
+######################################################################### 
+def main():
+    """main function"""
+    #
+    # Define default config file name
+    #
+    os_name = platform.system().lower()
+    if os_name == 'linux':
+        default_kms_copy_config_file = '/home/users/demo/key_backup.json'

```

Comment:
That should use the HOME directory of the actual user.

Reply:
Fixed